web/src/app/modules/actions/modules/export-testspecification-button/components/export-testspecification-button.component.ts

```diff
@@ -0,0 +1,181 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { TestSpecification } from '../../../../../model/TestSpecification';
+import { TestCase } from '../../../../../model/TestCase';
+import { TestParameter } from '../../../../../model/TestParameter';
+import { ParameterAssignment } from '../../../../../model/ParameterAssignment';
+import { Url } from '../../../../../util/url';
+import { Type } from '../../../../../util/type';
+import { IContainer } from '../../../../../model/IContainer';
+import { SpecmateDataService } from '../../../../data/modules/data-service/services/specmate-data.service';
+import { ValidationService } from '../../../../forms/modules/validation/services/validation.service';
+import { TranslateService } from '@ngx-translate/core';
+import { AuthenticationService } from '../../../../views/main/authentication/modules/auth/services/authentication.service';
+import { AccessRights } from '../../../../../model/AccessRights';
+import { UserToken } from '../../../../views/main/authentication/base/user-token';
+
+@Component({
+    moduleId: module.id.toString(),
+    selector: 'export-testspecification-button',
+    templateUrl: 'export-testspecification-button.component.html',
+    styleUrls: ['export-testspecification-button.component.css']
+})
+
+export class ExportTestspecificationButton {
+
+    @Input()
+    public testSpecification: TestSpecification;
+
+    private contents: IContainer[];
+    private testCases: TestCase[];
+    private testParameters: TestParameter[];
+    private parameterAssignments: ParameterAssignment[];
+    private finalCsvString: string;
+
+    constructor(
+        private dataService: SpecmateDataService,
+        private validation: ValidationService,
+        private translate: TranslateService,
+        private auth: AuthenticationService) { }
+
+    // Export Function
+    public exportTestSpecification(): void {
+        if (!this.enabled) {
+          return;
+        }
+
+        this.dataService.readContents(this.testSpecification.url)
```

Comment:
```suggestion
        const contents = await this.dataService.readContents(this.testSpecification.url);
        this.contents = contents;
```

Reply:
agree, but just before i submit the changes will this be completed (readContents) before executing (loadTestParameters) ??? because loadTestParameters depends on (this.contents) variable.

Reply:
Yes, this is exactly what the ```await```-keyword is for. It's basically syntactic sugar to avoid all these ```promise.then(...)```-call chains. So what it does (simplified) is, that it wraps the code after the statement into a ```then(code below)```-call.